std/stdio.d

```diff
@@ -725,7 +725,8 @@ $(D rawRead) always reads in binary mode on Windows.
     {
         import std.exception : enforce, errnoEnforce;
 
-        enforce(buffer.length, "rawRead must take a non-empty buffer");
+        if (!buffer.length)
+            enforce(false, "rawRead must take a non-empty buffer");
```

Comment:
I see.

Or change enforce to take a strict parameter (and thus get auto inlined) and have a lazyEnforce which takes a lazy parameter for those cases where the benefit of laziness outweights the benefit of inlining.

Really though you're right, the compiler should be able to infer that there is no cost to strictly evaluating msg if it is immutable.